lib/OrganizationsFieldArray/EditOrganizationCard.js

```diff
@@ -0,0 +1,229 @@
+import { AppIcon } from '@folio/stripes/core';
+import { FormattedMessage } from 'react-intl';
+import { Field } from 'redux-form';
+import {
+  Card,
+  IconButton,
+  Button,
+  Layout,
+  Col,
+  Label,
+  Select,
+} from '@folio/stripes/components';
+import { required } from '../validators';
+import OrganizationCard from './OrganizationCard';
+
+export default class EditOrganizationCard extends React.Component {
+  static propTypes = {
+    name: PropTypes.string,
+    index: PropTypes.string,
+    organization: PropTypes.object,
+    onMarkforDeletion: PropTypes.func,
+    onDeleteField: PropTypes.func,
+    onReplaceField: PropTypes.func,
+    roles: PropTypes.arrayOf(PropTypes.object),
+  };
+
+  renderOrgInfoHeaderEnd = () => {
+    const { name, index } = this.props;
+    return (
+      <Button
+        data-test-unlink-button
+        buttonStyle="danger"
+        id={`${name}-unlink-${index}`}
+        onClick={() => this.unlinkOrg()}
+      >
+        <FormattedMessage id="stripes-erm-components.organizations.unlinkOrganization" />
+      </Button>
+    );
+  }
+
+  renderHeaderStart = (isOrgPresent) => {
+    return (
+      isOrgPresent ?
+        (
+          <strong>
+            <FormattedMessage id="stripes-erm-components.organization" />
+          </strong>
+        ) : (
+          <span>
+            <AppIcon
+              app="organizations"
+              size="small"
+            />
+            {' '}
+            <strong>
+              <FormattedMessage id="stripes-erm-components.organization" />
+            </strong>
+          </span>
+        )
+    );
+  };
+
+  renderHeaderEnd = (isOrgPresent, id) => {
+    const { index, name } = this.props;
+    return isOrgPresent ?
+      this.renderOrgInfoHeaderEnd(id) :
+      this.renderOrganizationPicker(index, name);
+  }
+
+  renderOrgInfoHeaderStart = () => (
+    <strong>
+      <FormattedMessage id="stripes-erm-components.organization" />
+    </strong>
+  );
+
+  getOrganizationCardProps = () => {
+    const { index, organization : { org, id } } = this.props;
+    const isOrgPresent = org && !isEmpty(org);
+    const headerStart = this.renderHeaderStart(isOrgPresent);
+    const cardStyle = isOrgPresent ? 'positive' : 'negative';
+    const name = `${this.props.name}[${index}].org`;
+    const headerEnd = this.renderHeaderEnd(isOrgPresent, id);
+    const key = index;
+    const fieldId = `${this.props.name}-nameOrg-${index}`;
+    const validate = isOrgPresent ? [required, this.validateDuplicateEntries] : required;
+    const renderDefault = !isOrgPresent && this.renderDefaultLayout;
+
+    return {
+      headerStart,
+      cardStyle,
+      index,
+      name,
+      headerEnd,
+      key,
+      id:fieldId,
+      org,
+      validate,
+      renderDefault
+    };
+  };
+
+  renderOrganizationCard = () => (
+    <Field
+      data-test-org-name
+      component={OrganizationCard}
+      {...this.getOrganizationCardProps()}
```

Comment:
Is it really necessary to define this in a separate function? I don't see it being used anywhere else...